From 8ce1fafbda0d1ccb673945b2c5f067dbbecee8de Mon Sep 17 00:00:00 2001
From: Simon McVittie <smcv@debian.org>
Date: Wed, 25 May 2016 07:09:01 +0100
Subject: [PATCH 08/11] Update patches from upstream bug 71 to retain
 CAP_NET_ADMIN

This fixes unsharing the network:

    $ bwrap --unshare-net --ro-bind / / /bin/true; echo $?
    Can't create loopback device
    1
---
 debian/patches/Add-unshare-user-try.patch                             | 2 +-
 debian/patches/Make-setuid-no-unprivileged-user-namespaces-work.patch | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/debian/patches/Add-unshare-user-try.patch b/debian/patches/Add-unshare-user-try.patch
index 1053ad7..5bf2275 100644
--- a/debian/patches/Add-unshare-user-try.patch
+++ b/debian/patches/Add-unshare-user-try.patch
@@ -14,7 +14,7 @@ Bug: https://github.com/projectatomic/bubblewrap/pull/71
  1 file changed, 10 insertions(+)
 
 diff --git a/bubblewrap.c b/bubblewrap.c
-index 59407a7..6c4664e 100644
+index 7fdeb35..f9140f4 100644
 --- a/bubblewrap.c
 +++ b/bubblewrap.c
 @@ -148,6 +148,7 @@ usage (int ecode, FILE *out)
diff --git a/debian/patches/Make-setuid-no-unprivileged-user-namespaces-work.patch b/debian/patches/Make-setuid-no-unprivileged-user-namespaces-work.patch
index ec18f7e..e9a2328 100644
--- a/debian/patches/Make-setuid-no-unprivileged-user-namespaces-work.patch
+++ b/debian/patches/Make-setuid-no-unprivileged-user-namespaces-work.patch
@@ -35,7 +35,7 @@ index e4d250b..6e47b5f 100644
  endif
  
 diff --git a/bubblewrap.c b/bubblewrap.c
-index 1f9331a..59407a7 100644
+index 1f9331a..7fdeb35 100644
 --- a/bubblewrap.c
 +++ b/bubblewrap.c
 @@ -358,7 +358,7 @@ do_init (int event_fd, pid_t initial_pid)
@@ -43,7 +43,7 @@ index 1f9331a..59407a7 100644
  
  /* low 32bit caps needed */
 -#define REQUIRED_CAPS_0 (CAP_TO_MASK (CAP_SYS_ADMIN) | CAP_TO_MASK (CAP_SYS_CHROOT) | CAP_TO_MASK (CAP_NET_ADMIN))
-+#define REQUIRED_CAPS_0 (CAP_TO_MASK (CAP_SYS_ADMIN) | CAP_TO_MASK (CAP_SYS_CHROOT) | CAP_TO_MASK (CAP_SETUID) | CAP_TO_MASK (CAP_SETGID))
++#define REQUIRED_CAPS_0 (CAP_TO_MASK (CAP_SYS_ADMIN) | CAP_TO_MASK (CAP_SYS_CHROOT) | CAP_TO_MASK (CAP_NET_ADMIN) | CAP_TO_MASK (CAP_SETUID) | CAP_TO_MASK (CAP_SETGID))
  /* high 32bit caps needed */
  #define REQUIRED_CAPS_1 0
  
-- 
2.8.1

